aws/cloudformation/lambdas/marketing-router/index.js

```diff
@@ -0,0 +1,38 @@
+'use strict';
+
+const MARKETING_DOMAINS = {
+  'marketing-test': 'dev.marketing.dev-code.org',
+};
+
+const marketingPaths = {
+  // Add key-value pairs for each path that should be served by the CMS
+  // e.g. '/videos': true,
+}
+
+module.exports.handler = (event, context, callback) => {
+  const request = event?.Records?.[0]?.cf?.request;
+  const requestEnvHeader = request?.origin?.custom?.customHeaders?.['x-cdo-env']?.[0]?.value;
+  const marketingDomain = MARKETING_DOMAINS[requestEnvHeader];
+  const uri = request?.uri;
+
+  // Set CMS origin if the requested path matches
+  if (marketingPaths[uri]) {
+    request.origin = {
+      custom: {
+        domainName: marketingDomain,
+        port: 443,
+        protocol: 'https',
+        path: '',
+        sslProtocols: ['TLSv1.2'],
+        readTimeout: 30,
+        keepaliveTimeout: 5,
+        customHeaders: {},
+      },
+    };
+
+    // Update the Host header to match the new origin
+    request.headers['host'] = [{ key: 'Host', value: marketingDomain }];
+  }
+
+  callback(null, request);
```

Comment:
Because this Lambda's functionality is not core to the user experience for the old/existing marketing web application server (Pegasus), we may need to mitigate the risk of an unanticipated exception in this Lambda causing valid HTTP requests to Pegasus to be served an HTTP error response. Can we wrap the entire event handler in a try/catch block and any uncaught exceptions, we'll continue processing the original HTTP request as if the router Lambda did not exist?
```javascript
catch (error) {
    // Log the error to CloudWatch
    console.error('Error in Lambda@Edge function:', error);
    
    // Return the original request to CloudFront to continue normal processing
    callback(null, event.Records[0].cf.request);
  }
```

Reply:
We can explicitly catch exceptions that we anticipate and handle them appropriately.